### ISSUE TITLE
Full-text indexing fails for non-English-language users

### ISSUE THREAD

**Issue by AresDev** [couchbase/couchbase-lite-ios#983]
I've build the framework from master branch and added to my project as the wiki says(https://github.com/couchbase/couchbase-lite-ios/wiki/Building-Couchbase-Lite), I'm having problems with every single queries that uses CBLTextKey in its views, it shows me this:

2015-11-14 22:10:20.302 salesforce[1808:266898] Error calling sqlite3_step (1: unknown tokenizer) SQLITE_ERROR
2015-11-14 22:10:20.303 salesforce[1808:266898] DB Query:         CREATE VIRTUAL TABLE IF NOT EXISTS fulltext             USING fts4(content, tokenize=unicodesn "stemmer=spanish")
2015-11-14 22:10:20.304 salesforce[1808:266898] WARNING: Error initializing fts4 schema: Error Domain=SQLite Code=1 "unknown tokenizer" UserInfo={NSLocalizedDescription=unknown tokenizer}
2015-11-14 22:10:20.341 salesforce[1808:266898] WARNING: CouchbaseLite: Failed to rebuild views (imagesByProduct): 501
2015-11-14 22:10:20.342 salesforce[1808:266898] WARNING: Failed to update view index: 501

Could you tell me what I'm missing please?

Thanks!

**Comment by snej** [comment]
That's a strange error; I don't think anyone's reported it before. It looks as though the custom tokenizer we use for full-text indexing isn't getting registered with SQLite. (Thus the "unknown tokenizer" error.)

To diagnose this, please set a breakpoint in CBL_SQLiteStorage.m at the call to `register_unicodesn_tokenizer` (currently line 213). Then step into the function and see whether it finishes successfully or returns an error. (I just noticed that our call doesn't check the return value — we should fix that.) Let me know what you find.

**Comment by snej** [comment]
Oh, I should also mention that, if you don't need the very latest features, you can download a binary of the 1.1.1 release [from Couchbase](http://www.couchbase.com/nosql-databases/downloads), or you can check out the `release/1.1.1` branch of the Git repo and build that.

**Comment by AresDev** [comment]
Jens thank you for your quick answer, I just did what you said and it calls that function twice starting the app, both times it returns in line "sqlite3_finalize(pStmt);" it doesn`t show any error. One thing I noticed is that it runs ok on simulator(iOS 9.1) but fails on device(Ipad mini iOS 9.1).

I need this version because the one on the release 1.1.1 takes forever to sync and crashes my app due to memory(https://github.com/couchbase/couchbase-lite-ios/issues/502)

I hope this helps.

**Comment by snej** [comment]
This must be a build error of some kind. You can download an automated build of the latest master branch [here](http://mobile.jenkins.couchbase.com/job/couchbase-lite-ios-builds/). These builds have passed our unit tests.

**Comment by AresDev** [comment]
Thank you Jens, I'll try that.

**Comment by snej** [comment]
A few days ago, by coincidence, I found a bug in our FTS tokenizer that causes this. The bug is that the non-English stemmers aren't being enabled, so attempting to use them causes an error initializing the tokenizer. (`USING fts4(content, tokenize=unicodesn "stemmer=spanish")`.)

I've fixed the problem in a branch, but I need to get it into the 1.2 release too or FTS will fail for non-English users.

**Comment by AresDev** [comment]
Thank you!!! Glad that you could replicated and fixed this error!

**Comment by snej** [comment]
If you want to try the fix: check out CBL (either release/1.1.1 or release/1.2 branch), then

```
cd vendor/CBForest/vendor/sqlite3-unicodesn
git checkout master
git pull
```
Then rebuild.

**Comment by snej** [comment]
Fixed on release/1.2 branch.
